edit_course/views.py

```diff
@@ -324,21 +329,122 @@ def get_form_kwargs(self):
 
     def form_valid(self, form):
         from .operations.clone import clone
-        instance = clone(
+        new_instance = clone(
             instance=self.instance,
             url=form.cleaned_data['url'],
+            name=form.cleaned_data['instance_name'],
             clone_teachers=form.cleaned_data['teachers'],
             clone_assistants=form.cleaned_data['assistants'],
             clone_usertags=form.cleaned_data['usertags'],
-            clone_categories=form.cleaned_data['categories'],
-            clone_modules=form.cleaned_data['modules'],
-            clone_chapters=form.cleaned_data['chapters'],
-            clone_exercises=form.cleaned_data['exercises'],
             clone_menuitems=form.cleaned_data['menuitems'],
             siskey=form.cleaned_data.get('sis'),
         )
+
+        if not all([settings.GITMANAGER_URL, form.cleaned_data.get('git_origin'), form.cleaned_data.get('git_branch')]):
+            # Do not create a new entry in Git manager
+            messages.success(self.request, _('COURSE_INSTANCE_CLONED'))
+            return self.redirect(new_instance.get_url('course-details'))
+
+        key = "{}_{}{}".format(self.course.code, form.cleaned_data['key_year'], form.cleaned_data['key_month'])
```

Comment:
We had some discussions about this before, would you want the whole key to be in lower case? The month/semester included?
Also, I used ``.format()`` because f string made the format of the key hard to see from the code because of the long variable names.

Reply:
Yes, we agreed that it can be like the way it is currently. Also because a lot of the existing keys in Git manager already have this format.